### PR TITLE
Feat/variable aggregator impl

### PR DESCRIPTION
**Pull request by disaster1-tesk**
### Describe what this PR does / why we need it
add variableaggregatorNodeDataConverter and generics for support new node

### Does this pull request fix one issue?
None

### Describe how you did it
add variableaggregatorNodeDataConverter and variableaggregatorNode  impl

### Describe how to verify it
com.alibaba.cloud.ai.api.DSLAPI Import/export dsl

### CODE REVIEW

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    groupMap.put("groupId",group.getGroupId());
+                    groups.add(groupMap);
+                }
+                advancedSettings.put("groups",groups);
+                result.put("advanced_settings",advancedSettings);
+            }else {
+                result.put(field.getName(), value);
+            }
+        });
+        return result;
```
Comment (chickenlj) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java:71]: Would it be simpler if we use Jackson to do the dump convert here?

Comment (disaster1-tesk) [reply spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java:71]: Dify-compatible data types require the use of map to load data

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/nodedata/VariableAggregatorNodeData.java
Comment (robocanic) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/nodedata/VariableAggregatorNodeData.java:1]: I got some issues maybe useful:
1. `type`, `title`, `desc`, `selected` these fields are common fields which has been processed by the Node, so these fields are redundant.
2. if fields naming is in `Camel Case` style will be better?

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public VariableAggregatorNodeData parseDifyData(Map<String, Object> data) {
+        VariableAggregatorNodeData variableAggregatorNodeData = new VariableAggregatorNodeData(null, null);
+        variableAggregatorNodeData.setTitle((String) data.get("title"));
+        variableAggregatorNodeData.setType((String) data.get("type"));
+        variableAggregatorNodeData.setSelected((Boolean) data.get("selected"));
+        variableAggregatorNodeData.setOutput_type((String) data.get("output_type"));
+        variableAggregatorNodeData.setDesc((String) data.get("desc"));
+        variableAggregatorNodeData.setVariables((List<List<String>>) data.get("variables"));
+        Map<String,Object> advanced_settings = (Map<String, Object>) data.get("advanced_settings");
+        VariableAggregatorNodeData.AdvancedSettings advancedSettings  = new VariableAggregatorNodeData.AdvancedSettings();
```
Comment (robocanic) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java:36]: Maybe multiple `setXXX`  calls  can be simplified as `setTitle().setType().setXXX()`?

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/NodeData.java
Comment (robocanic) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/NodeData.java:1]: image
`type`, `title`, `desc`, `selected`, these fields has been stated and processed by Node, no need to restate

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/nodedata/VariableAggregatorNodeData.java
Comment (robocanic) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/model/workflow/nodedata/VariableAggregatorNodeData.java:1]: Keep `Camel Case` style would be better

File: spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                .advanced_settings(advancedSettings)
+                .build();
+    }
+
+
+    @Override
+    @SneakyThrows
+    public Map<String, Object> dumpDifyData(VariableAggregatorNodeData nodeData) {
+        HashMap<String, Object> result = new HashMap<>();
+        ReflectionUtils.doWithFields(nodeData.getClass(), field -> {
```
Comment (robocanic) [inline spring-ai-alibaba-graph/spring-ai-alibaba-graph-studio/src/main/java/com/alibaba/cloud/ai/service/dsl/nodes/VariableAggregatorNodeDataConverter.java:47]: Why we need to use reflect here?

**Review by robocanic — APPROVED** [review]
LGTM

**Pull request merged**